flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/physical/batch/DynamicPartitionPruningRule.java

```diff
@@ -152,9 +152,39 @@ private static List<Integer> getAcceptedFieldIndices(
             }
         }
 
-        return acceptedFields.stream()
-                .map(f -> factScan.getRowType().getFieldNames().indexOf(f))
-                .collect(Collectors.toList());
+        if (factCalc == null) {
+            return acceptedFields.stream()
+                    .map(f -> factScan.getRowType().getFieldNames().indexOf(f))
+                    .collect(Collectors.toList());
+        } else {
+            return getAcceptedFieldsIndicesInCalc(acceptedFields, factJoinKeys, factCalc, factScan);
+        }
+    }
+
+    private static List<Integer> getAcceptedFieldsIndicesInCalc(
+            @Nullable List<String> acceptedFields,
+            List<Integer> factJoinKeys,
+            BatchPhysicalCalc factCalc,
+            BatchPhysicalTableSourceScan factScan) {
+        if (acceptedFields == null) {
```

Comment:
acceptedFields should not be null, otherwise NPE will be thrown at line 145. So we should add Null check at line 145

Reply:
> acceptedFields should not be null, otherwise NPE will be thrown at line 145. So we should add Null check at line 145

Done!